graphviz/jupyter_integration.py

```diff
@@ -1,12 +1,70 @@
 """Display rendered graph as SVG in Jupyter Notebooks and QtConsole."""
-
 from . import piping
 
 __all__ = ['JupyterSvgIntegration']
 
+_default_jupyter_representation = 'image/svg+xml'
+
+_jupyter_repr_functions = {
+    'image/svg+xml': lambda graph: graph.pipe(format='svg',
+                                              encoding=graph._encoding),
+    'image/png': lambda graph: graph.pipe(format='png'),
+    'image/jpeg': lambda graph: graph.pipe(format='jpeg'),
+
+    # The following does not work yet. Not clear why, because the documentation
+    # has only few information about them.
+    # 'application/json': lambda graph: json.loads(
+    #     graph.pipe(format='json', encoding=graph._encoding)),
+    # 'application/pdf': lambda graph: graph.pipe(format='pdf'),
+
+    # No reasonable representation
+    # 'text/plain': None,
+    # 'text/latex': None,
+    # 'text/html': None,
+    # 'application/javascript': None,
+    # 'text/markdown': None,
+}
+
+JUPYTER_REPRESENTATIONS = set(_jupyter_repr_functions.keys())
+
+
+def set_default_jupyter_representation(jupyter_representation: str) -> str:
+    """Change the default jupyter_representation"""
+    global _default_jupyter_representation
+
+    if jupyter_representation not in JUPYTER_REPRESENTATIONS:
+        similar = {
+            r
+            for r in JUPYTER_REPRESENTATIONS
+            if jupyter_representation.lower() in r}
+        if similar:
+            raise ValueError(
+                f'unknown jupyter representation: {jupyter_representation!r}\n'
+                f'Suggestions: {similar}')
+        else:
```

Comment:
Sure, I can make it case-insensitive, also I don't see here a motivation, because the `mime` names aren't obvious.
My idea here was, that the user might request `png` and then gets the suggestion, that the correct values would be `image/png` or in case of svg you need the xml postfix (`image/svg+xml`).

(Normally I use the difflib as in [Dispatcher](https://github.com/fgnt/paderbox/blob/master/paderbox/utils/mapping.py) to get informative exceptions. It saved me much time for debugging.)

Reply:
On a second thought, maybe case-insensitive would already be overkill.

Supporting `image/svg` as an alias for `image/svg+xml' might help if something in Jupyter changes.

> My idea here was, that the user might request `png`

Just in case: In my understanding this method is not to be called by users.

Reply:
Sorry for my confusion: the function`set_default_format` is indeed for users as you say. I was thinking about `_repr_mimebundle_`: providing hints SG for the former (if you think its worth implementing) but not for the latter is what I meant to say (e.g. we could call the `verify` function in the former and then provide hints in that function on ValueError). HTH

Reply:
Ah, ok. Nevertheless, your suggestion is more common in python.